airbyte-integrations/connectors/source-twilio-taskrouter/source_twilio_taskrouter/twilio_taskrouter.yaml

```diff
@@ -0,0 +1,75 @@
+version: "0.1.0"
+
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "https://taskrouter.twilio.com"
+    http_method: "GET"
+    authenticator:
+      type: BasicHttpAuthenticator
+      username: "{{ config['account_sid'] }}"
+      password: "{{ config['auth_token'] }}"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  allworkspaces_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "allworkspaces"
+      path: "/v1/Workspaces"
+      primary_key: "sid"
+    retriever:
+      $ref: "*ref(definitions.base_stream.retriever)"
+      record_selector:
+        extractor:
+          field_pointer: ["workspaces"]
+      paginator:
+        type: "DefaultPaginator"
+        page_size_option:
+          inject_into: "request_parameter"
+          field_name: "page_size"
+        pagination_strategy:
+          type: "CursorPagination"
+          cursor_value: "{{ headers['url']['next'] }}"
```

Comment:
you try to tell the paginator to get the cursor value from the headers. Are you sure the headers contain the value you want? 

The following is an example response header when I try to GET `/Workers`
```
Date: Mon, 07 Nov 2022 04:10:22 GMT
Content-Type: application/json
Content-Length: 4151
Connection: keep-alive
x-rate-limit-remaining: 50
x-rate-limit-limit: 50, 50;window=1
Twilio-Concurrent-Requests: 1
x-rate-limit-config: Worker-List
Twilio-Request-Id: RQ5982d0470f1e214bfd748a8eb699f817
Twilio-Request-Duration: 0.021
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: Accept, Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since, Idempotency-Key
Access-Control-Allow-Methods: GET, POST, DELETE, OPTIONS
Access-Control-Expose-Headers: ETag
Access-Control-Allow-Credentials: true
X-Powered-By: AT-5000
X-Shenanigans: none
X-Home-Region: us1
X-API-Domain: taskrouter.twilio.com
Strict-Transport-Security: max-age=31536000
```

Reply:
Okay, I'll figure out the appropriate cursor value to solve the response header

Reply:
the cursor value doesn't come from the header. It's part of the response body, it's body["meta"]["next_page_uri"]

Reply:
I had tried that but it shows that body is not defined

Reply:
I'm trying with response['meta']['next_page_url'] but it shows the JSON Decode Error at line 1 col 1 char(0)

Reply:
there's a case that next_page_url is null, which indicates you have reached the last page. And maybe that's the error reason of JSON deserializer.

Reply:
When i try checking the API, it does show the next page url. Could it maybe related to the page_token injecting the path?
![image](https://user-images.githubusercontent.com/59562284/201540397-822a69b1-bc0e-4ea5-96b5-75b65e7a49b2.png)

Reply:
or maybe it's because I am using a record extractor to filter out the meta json respone?

Reply:
> When i try checking the API, it does show the next page url. Could it maybe related to the page_token injecting the path? ![image](https://user-images.githubusercontent.com/59562284/201540397-822a69b1-bc0e-4ea5-96b5-75b65e7a49b2.png)

The next_page_url is the URL for the next page, you will get another next_page_url in the meta with this URL, until the value is null. This is the pseudo-code I provided you above. The right pagination strategy or paginator in the YAML works like the pseudo-code if it's configured correctly. You can also manually change the URL in a tool like postman to understand how it works. 

```
request_url = "base_url/v1/workers";
do {
  response = request.get(request_url);
  request_url = response.body.meta.next_page_url;
} while (request_url != null)
```

Reply:
I did some adjustments and now it reads all the records, with the page size being 50

/home/alcadeus/Pictures/Screenshots/Screenshot from 2022-11-15 12-02-46.png